src/cli/commands/add.js

```diff
@@ -174,6 +175,11 @@ export async function run(
   }
 
   const lockfile = await Lockfile.fromDirectory(config.cwd, reporter);
+
+  await executeLifecycleScript(config, 'preinstall');
+
   const install = new Add(args, flags, config, reporter, lockfile);
   await install.init();
+
+  await executeLifecycleScript(config, 'postinstall');
```

Comment:
We'll need to run the `install` lifecycle script and `prepublish` one here too, you can see `commands/install.js`.